pkg/analysis/passes/checksum/checksum_test.go

```diff
@@ -0,0 +1,277 @@
+	}
+	_, err := Analyzer.Run(pass)
+	assert.NoError(t, err)
+
+	require.Len(t, interceptor.Diagnostics, 0)
+}
+
+func TestInvalidMD5Checksum(t *testing.T) {
+	var interceptor testpassinterceptor.TestPassInterceptor
+
+	pass := &analysis.Pass{
+		ResultOf: map[*analysis.Analyzer]interface{}{},
+		CheckParams: &analysis.CheckParams{
+			Checksum:              "f9b1c42c45cbf4953d7da5c31b3d73d9",
+			ArchiveCalculatedMD5:  "f9b1c42c45cbf4953d7da5c31b3d73d4",
+			ArchiveCalculatedSHA1: "187cd80948f240957bf8399745335b89f005c5f0",
+		},
+		Report: interceptor.ReportInterceptor(),
+	}
+
+	_, err := Analyzer.Run(pass)
+	assert.NoError(t, err)
+
+	// should not call the report function
+	require.Len(t, interceptor.Diagnostics, 1)
+	require.Equal(
+		t,
+		"The provided checksum f9b1c42c45cbf4953d7da5c31b3d73d9 does not match the plugin archive",
+		interceptor.Diagnostics[0].Title,
+	)
+
+}
+
+func TestInvalidSHA1Checksum(t *testing.T) {
+	var interceptor testpassinterceptor.TestPassInterceptor
+
+	pass := &analysis.Pass{
+		ResultOf: map[*analysis.Analyzer]interface{}{},
+		CheckParams: &analysis.CheckParams{
+			Checksum:              "187cd80948f240957bf8399745335b89f005c5f0",
+			ArchiveCalculatedMD5:  "f9b1c42c45cbf4953d7da5c31b3d73d9",
+			ArchiveCalculatedSHA1: "187cd80948f240957bf8399745335b89f005c5f1",
+		},
+		Report: interceptor.ReportInterceptor(),
+	}
+
+	_, err := Analyzer.Run(pass)
+	assert.NoError(t, err)
+
+	// should not call the report function
+	require.Len(t, interceptor.Diagnostics, 1)
+	require.Equal(
+		t,
+		"The provided checksum 187cd80948f240957bf8399745335b89f005c5f0 does not match the plugin archive",
+		interceptor.Diagnostics[0].Title,
+	)
+}
+
+func TestSkipsCheckSumWhenNoProvided(t *testing.T) {
+	var interceptor testpassinterceptor.TestPassInterceptor
+
+	pass := &analysis.Pass{
+		ResultOf: map[*analysis.Analyzer]interface{}{},
+		CheckParams: &analysis.CheckParams{
+			Checksum:              "", // no checksum
+			ArchiveCalculatedMD5:  "f9b1c42c45cbf4953d7da5c31b3d73d9",
+			ArchiveCalculatedSHA1: "187cd80948f240957bf8399745335b89f005c5f0",
+		},
+		Report: interceptor.ReportInterceptor(),
+	}
+
+	_, err := Analyzer.Run(pass)
+	assert.NoError(t, err)
+
+	// should not call the report function
+	require.Len(t, interceptor.Diagnostics, 0)
+}
+
+func TestValidMD5ChecksumFromUrl(t *testing.T) {
+	var interceptor testpassinterceptor.TestPassInterceptor
+
+	pass := &analysis.Pass{
+		ResultOf: map[*analysis.Analyzer]interface{}{},
+		CheckParams: &analysis.CheckParams{
+			Checksum:              "https://example.com/checksum",
+			ArchiveCalculatedMD5:  "f9b1c42c45cbf4953d7da5c31b3d73d9",
+			ArchiveCalculatedSHA1: "187cd80948f240957bf8399745335b89f005c5f0",
+		},
+		Report: interceptor.ReportInterceptor(),
+	}
+
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	httpmock.RegisterResponder("GET", "https://example.com/checksum",
+		httpmock.NewStringResponder(200, "f9b1c42c45cbf4953d7da5c31b3d73d9 "))
```

Comment:
Nit: Lint :smiley: 

```suggestion
	httpmock.RegisterResponder("GET", "https://example.com/checksum",
	httpmock.NewStringResponder(200, "f9b1c42c45cbf4953d7da5c31b3d73d9 "))
```

(we should probably add a `golangci-lint` step to the CI pipeline for PRs in this repo, I can open an issue for that)

Reply:
This is part of https://github.com/segmentio/golines to make long lines more readable. All the code is already well formatted with gofmt too